systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/StrimziPodSetUtils.java

```diff
@@ -52,6 +53,82 @@ public static void waitForStrimziPodSetLabelsDeletion(String namespaceName, Stri
         }
     }
 
+    /**
+     * Waits until the single specified annotation of a Pod changes its value.
+     *
+     * @param namespaceName    the namespace in which the Pod is located.
+     * @param podName          the name of the Pod whose annotation value needs to be checked.
+     * @param strimziPodSetName the name of the SPS whose Pod's annotation is being observed.
+     * @param annotationKey    the key of the annotation whose value needs to be checked.
+     * @param annotationValue  the expected value of the annotation to change from.
+     */
+    public static void waitUntilPodAnnotationChange(String namespaceName, String podName, String strimziPodSetName, String annotationKey, String annotationValue) {
+
+        LOGGER.info("Waiting for Pod: {}/{} to change annotation value: {}-{}", namespaceName, podName, annotationKey, annotationValue);
+        final ObjectMapper objectMapper = new ObjectMapper();
+
+
+        TestUtils.waitFor("Pod to change value of annotation" + annotationKey + " with key " + annotationKey, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+            DELETION_TIMEOUT, () -> {
+                // parse pod from SPS
+                StrimziPodSet sps = StrimziPodSetResource.strimziPodSetClient().inNamespace(namespaceName).withName(strimziPodSetName).get();
+                var podMap = sps.getSpec().getPods();
+                Pod[] spsPods = objectMapper.convertValue(podMap, Pod[].class);
+                Pod pod = Arrays.stream(spsPods).filter(p -> p.getMetadata().getName().contains(podName)).findAny().get();
+
+                return !annotationValue.equals(pod.getMetadata().getAnnotations().get(annotationKey));
+            });
+        LOGGER.info("Pod: {}/{} changed annotation value: {}-{}", namespaceName, podName, annotationKey, annotationValue);
+    }
+
+    /**
+     * Waits for a specific annotation key-value pair to remain stable (from start to beginning) on a given Pod.
+     *
+     * @param namespaceName   the namespace in which the Pod is located.
+     * @param annotationKey   the key of the annotation to be checked.
+     * @param annotationValue the expected value of the annotation to be checked.
+     * @param strimziPodSetName the name of the SPS whose Pod's annotation is being observed.
+     * @param podName the name of the Pod whose annotation is being observed.
+     */
+    public static void waitForPrevailedPodAnnotationKeyValuePairs(String namespaceName, String annotationKey, String annotationValue, String strimziPodSetName, String podName) {
+        LOGGER.info("Waiting for label: {}-{} to keep its value in namespace: {}", namespaceName, annotationKey, annotationValue);
```

Comment:
I guess you have wrong order of the parameters, or?
`{}-{}` should be key/value and `in namespace: {}` should be the namespace right?

Reply:
yes, should be like this
```suggestion
        LOGGER.info("Waiting for label: {}-{} to keep its value in namespace: {}", annotationKey, annotationValue, namespaceName);
```